### ISSUE TITLE
Few notes about fiber adapter

### ISSUE THREAD

**Issue by x-user** [danielgtaylor/huma#345]
If i understand correctly `fiberCtx.SetReadDeadline` actually doesn't work and probably will not. Also if at same time we will add ability to get client IP address from `huma.Context` it will not work with current fiber adapter implementation.

Currently `fiberAdapter.ServeHTTP` calls `App.Test` to handle requests. This method uses `testConn` to mock incoming connection.

But `testConn.Set*Deadline` methods do nothing:
```go
func (*testConn) SetDeadline(_ time.Time) error      { return nil }
func (*testConn) SetReadDeadline(_ time.Time) error  { return nil }
func (*testConn) SetWriteDeadline(_ time.Time) error { return nil }
```
And `testConn.*Addr` methods return hardcoded values:
```go
func (*testConn) LocalAddr() net.Addr                { return &net.TCPAddr{Port: 0, Zone: "", IP: net.IPv4zero} }
func (*testConn) RemoteAddr() net.Addr               { return &net.TCPAddr{Port: 0, Zone: "", IP: net.IPv4zero} }
```

**Comment by excavador** [comment]
@x-user hello!

I am maintaing humafiber adapter (not officially).
I read your issue and comment, and I want to drag your attention to the following:
- fasthttp and fiber has the very special way to deal with content (because of zero-allocation)
- fiber.Ctx.Context and fiber.Ctx.UserContext works in the very specific way.

The long story short - to get expected behavior with Deadline, you need to alter "UserContext" inside middleware and you should use "UserContext" with altering to get it.

Take a look to this article from fiber documentation about timeout
- https://docs.gofiber.io/api/middleware/timeout/

Also you could take a look to this pull request with fixes to huma.Context <=> Fiber integration
- https://github.com/danielgtaylor/huma/pull/725

*Bottom line* - use fiber "Timeout" middleware + fixed version of humafiber from my PR, after that you will receive the expected behavior with timeout

Please ask questions if you have any

**Comment by wolveix** [comment]
It sounds like this has been addressed, but please feel free to re-open if not. Thanks!
